packages/opentelemetry-exporter-collector/src/platform/browser/sendSpans.ts

```diff
@@ -0,0 +1,148 @@
+/*!
+ * Copyright 2019, OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { hrTime, hrTimeToTimeStamp } from '@opentelemetry/core';
+import { Logger } from '@opentelemetry/types';
+import { CollectorExporter } from '../../CollectorExporter';
+import * as collectorTypes from '../../types';
+
+/**
+ * function that is called once when {@link ExporterCollector} is initialised
+ * @param shutdownF shutdown method of {@link ExporterCollector}
+ */
+export function onInit(shutdownF: EventListener) {
+  window.addEventListener('unload', shutdownF);
+}
+
+/**
+ * function to be called once when {@link ExporterCollector} is shutdown
+ * @param shutdownF - shutdown method of {@link ExporterCollector}
+ */
+export function onShutdown(shutdownF: EventListener) {
+  window.removeEventListener('unload', shutdownF);
+}
+
+/**
+ * function to send spans to the [opentelemetry collector]{@link https://github.com/open-telemetry/opentelemetry-collector}
+ *     using the standard http/https node module
+ * @param spans
+ * @param onSuccess
+ * @param onError
+ * @param collectorExporter
+ */
+export function sendSpans(
+  spans: collectorTypes.Span[],
+  onSuccess: () => void,
+  onError: (status?: number) => void,
+  collectorExporter: CollectorExporter
+) {
+  const exportTraceServiceRequest: collectorTypes.ExportTraceServiceRequest = {
+    node: {
+      identifier: {
+        hostName: collectorExporter.hostName || window.location.host,
+        startTimestamp: hrTimeToTimeStamp(hrTime()),
+      },
+      libraryInfo: {
+        language: collectorTypes.LibraryInfoLanguage.WEB_JS,
+        // coreLibraryVersion: , not implemented
```

Comment:
I think we should export the core library version and use it here, something like this: https://github.com/census-instrumentation/opencensus-node/blob/master/packages/opencensus-core/src/common/version.ts

Reply:
something doesn't work correctly, 
`Can't resolve '../package.json' in ../opentelemetry-js/packages/opentelemetry-core/build/src/common'`  
it looks like this
```
try {
  pjson = require('../../../package.json');
} catch {
  pjson = require('../../package.json');
}
```
but it fails either in core tests or then in next class that is using this.
It looks like both `require` are executed and even though the catch is there it is still failing

Reply:
might be related to `karma-coverage-istanbul-reporter`

Reply:
Summarising so in node.js it works fine, but not in web :/. 
So far I tried different approaches and for web it will always show error even though I made the karma not to fail. Because of that my last idea is to create a script that could replace this value in runtime when the package is built so it will simply contain `export const version = '0.2.0'`;

Reply:
I think the right solution is to have a `version.ts` file in the `core` package that exports the version and then just import that constant. We will need to have some notes in the releasing process that you need to update that file, and maybe even make it eventually be part of the `npm run bump` command.

We did this for OpenCensus Web, see this [version.ts file](https://github.com/census-instrumentation/opencensus-web/blob/14d9c91b58d7c314f392f76e886773174b26ceae/packages/opencensus-web-core/src/version.ts), and its [use in the exporter code](https://github.com/census-instrumentation/opencensus-web/blob/81361d220f4dc37cae7e2c22ad8b0bc025f1ad9c/packages/opencensus-web-exporter-ocagent/src/ocagent.ts#L116).

Reply:
Replacing the value at run time when the package is built is a clever idea. It feels more complex then just exporting a constant and telling people to update it when new releases come out though ....

Reply:
We could potentially abuse the `const enum` mechanism to our benefit here.

In core we could have a `constants.ts` file:

```typescript
export const enum Constants {
  VERSION = "0.2.0";
}
```

then in other files you import it and use it like this:

```typescript
import { Constants } from "@opentelemetry/core";

console.log(Constants.VERSION);
```

which actually compiles to this behind the scenes:

```javascript
"use strict";
exports.__esModule = true;
console.log("1" /* VERSION */);
```

Reply:
Oh wow, `const enum` is a smart idea here! I really like it :)

It fits right along with https://github.com/open-telemetry/opentelemetry-js/issues/538